hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -427,10 +427,18 @@ def scalar_dtypes():
 
 def defines_dtype_strategy(strat):
     # type: (T) -> T
+    def convert_to_dtype(x):
+        """Helper to debug issue #1798."""
+        try:
+            return np.dtype(x)
+        except ValueError:  # pragma: no cover
+            note("Got invalid dtype value=%r from strategy=%r" % (x, strat))
```

Comment:
It would be nice to have this tested. You could test it by defining a deliberately invalid dtype strategy in a test and using `defines_dtype_strategy` on it.